File: StudioProject/SimpleGithubApp/app/src/main/java/skim/dev/kr/simplegithubapp/api/AuthApi.java
package skim.dev.kr.simplegithubapp.api;

import android.support.annotation.NonNull;

import retrofit2.Call;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.Headers;
import retrofit2.http.POST;
import skim.dev.kr.simplegithubapp.api.model.GithubAccessToken;

public interface AuthApi {

    @FormUrlEncoded
    @POST("login/oauth/access_token")
    @Headers("Accept: application/json")
    Call<GithubAccessToken> getAccessToken(
            @NonNull @Field("client_id") String clientId,
            @NonNull @Field("client_secret") String clientSecret,
            @NonNull @Field("code") String code);
}

File: StudioProject/SimpleGithubApp/app/src/main/kotlin/skim/dev/kr/simplegithubapp/api/AuthApi.kt
package skim.dev.kr.simplegithubapp.api

import io.reactivex.Observable
import retrofit2.http.Field
import retrofit2.http.FormUrlEncoded
import retrofit2.http.Headers
import retrofit2.http.POST
import skim.dev.kr.simplegithubapp.api.model.GithubAccessToken

interface AuthApi {

    @FormUrlEncoded
    @POST("login/oauth/access_token")
    @Headers("Accept: application/json")
    fun getAccessToken(
            @Field("client_id") clientId: String,
            @Field("client_secret") clientSecret: String,
            @Field("code") code: String): Observable<GithubAccessToken>

}
